packages/resolver/lib/sources/vyper.ts

```diff
@@ -0,0 +1,144 @@
+import debugModule from "debug";
+const debug = debugModule("resolver:sources:vyper");
+import path from "path";
+import { ContractObject } from "@truffle/contract-schema/spec";
+import { ResolverSource, SourceResolution } from "../source";
+
+export class Vyper implements ResolverSource {
+  contractsDirectory: string;
+  wrappedSources: ResolverSource[];
+  //because this ResolverSource has to do an actual resolution just to
+  //do a resolveDependencyPath, I'm giving it a cache to prevent redoing
+  //the work of resolution later
+  cache: {
+    [filePath: string]: SourceResolution
+  };
+
+  constructor(wrappedSources: ResolverSource[], contractsDirectory: string) {
+    this.wrappedSources = wrappedSources;
+    this.cache = {};
+    this.contractsDirectory = contractsDirectory;
+  }
+
+  require(): ContractObject | null {
+    //out of scope for this resolver source
+    return null;
+  }
+
+  async resolve(importModule: string, importedFrom: string) {
+    importedFrom = importedFrom || "";
+
+    debug("importModule: %s", importModule);
+    debug("importedFrom: %s", importedFrom);
+
+    //attempt to just resolve as if it's a file path rather than Vyper module
+    for (const source of this.wrappedSources) {
+      const directlyResolvedSource = await source.resolve(importModule, importedFrom);
+      if (directlyResolvedSource.body) {
+        debug("found directly");
+        return directlyResolvedSource;
+      }
+    }
+    //otherwise, it's time for some Vyper module processing...
+
+    //only attempt this if what we have looks like a Vyper module
+    if (!importModule.match(/^[\w.]+$/)) {
+      debug("clearly not a Vyper module");
+      return { body: undefined, filePath: undefined };
+    }
+
+    const importPath = moduleToPath(importModule); //note: no file extension yet
+    debug("importPath: %s", importPath);
+    const explicitlyRelative = importModule[0] === ".";
```

Comment:
Is there any reason to add the same check as to whether it starts with "./" or "../" like in the other utility method mentioned?

Reply:
Good question! :)  The answer is that this is actually deliberate, because we're checking here the unprocessed `importModule`, rather than the converted `importPath`.  Whereas `importPath` starting with a `.` could just mean a hidden file or directory, `importModule` starting with a `.` definitely indicates a relative path (see the code below for `moduleToPath`).

Maybe I should add a comment pointing this out.

Reply:
(Comment has since been added. :) )